### ISSUE TITLE
markdown2 is slow

### ISSUE THREAD

**Issue by techtonik** [trentm/python-markdown2#154]
```
Parsing the Markdown Syntax document 1000 times...
Mistune: 12.9425s
Misaka: 0.537176s
Markdown: 47.7091s
Markdown2: 80.5163s
cMarkdown: 0.680664s
```

http://lepture.com/en/2014/markdown-parsers-in-python

**Comment by marios-zindilis** [comment]
`python-markdown2` is slow because Python's `re` library is slow, and `python-markdown2` uses it a lot. I made some quick and very unscientific tests with replacing some methods with equivalents that don't use `re`, and they were faster. If the maintainers of `python-markdown2` want, I can try to send some pull requests. 

I documented some results [here](https://zindilis.com/blog/2015/11/07/regex-in-python-markdown2-pt1.html) and [here](https://zindilis.com/blog/2015/11/22/regex-in-python-markdown2-pt2.html).
